File: kube/src/main/java/com/jacksonrakena/infrastructure/util/SecretUtils.kt
package com.jacksonrakena.infrastructure.util

import org.cdk8s.plus28.TlsSecretProps
import java.nio.file.Files
import java.nio.file.Path

fun loadTlsSecretFromFolder(path: String): TlsSecretProps {
    val crt = Files.readString(Path.of(path).resolve("tls.crt"))
    val key = Files.readString(Path.of(path).resolve("tls.key"))

    return TlsSecretProps.builder().tlsCert(crt).tlsKey(key).build()
}

File: src/util/secret-utils.ts
import * as fs from "fs";
import * as path from "path";
import * as kplus from "cdk8s-plus-28";

export function loadTlsSecretFromFolder(
  folderPath: string
): kplus.TlsSecretProps {
  const crt = fs.readFileSync(path.join(folderPath, "tls.crt"), "utf-8");
  const key = fs.readFileSync(path.join(folderPath, "tls.key"), "utf-8");
  return { tlsCert: crt, tlsKey: key };
}
